build.assets/robotest/pr_config.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+set -o errexit
+set -o nounset
+set -o pipefail
+
+source $(dirname $0)/utils.sh
+
+# UPGRADE_MAP maps gravity version -> list of linux distros to upgrade from
+declare -A UPGRADE_MAP
+UPGRADE_MAP[$(recommended_upgrade_tag $(branch 7.0.x))]="centos:7 debian:9 ubuntu:18" # compatible LTS version
+UPGRADE_MAP[7.0.0]="ubuntu:16"
+# 6.2 and 6.3 ignored in PR builds per https://github.com/gravitational/gravity/pull/1760#pullrequestreview-437838773
+# UPGRADE_MAP[$(recommended_upgrade_tag $(branch 6.3.x))]="redhat:7" # compatible non-LTS version
+# UPGRADE_MAP[6.3.0]="ubuntu:16"  # disabled due to https://github.com/gravitational/gravity/issues/1009
+# UPGRADE_MAP[$(recommended_upgrade_tag $(branch 6.2.x))]="redhat:7" # compatible non-LTS version
+# UPGRADE_MAP[6.2.0]="ubuntu:16"
+
+UPGRADE_VERSIONS=${!UPGRADE_MAP[@]}
```

Comment:
Can we dump the entire constructed upgrade matrix to CLI (unless we already do that)? Just as a log, to be able to see calculated versions in the job output, in case the algorithm malfunctions. Same for nightly config.

Reply:
We already do that:

https://github.com/gravitational/gravity/blob/d4b3c67b2c8f785883fb1581c36e5ab435a6f71f/build.assets/robotest_run_suite.sh#L111

The above was annoying to test though, as it required adding/remvoing an exit call to only view the config without executing it.

This changset makes the configs easier to check in an adhoc manner.  See following, from "Testing done":

<details><summary>./build.assets/robotest/pr_config.sh</summary>

```
walt@work:~/git/gravity$ ./build.assets/robotest/pr_config.sh     
install={"flavor":"three","nodes":3,"role":"node","os":"redhat:7","storage_driver":"overlay2"}
install={"installer_url":"/installer/opscenter.tar","nodes":1,"flavor":"standalone","role":"node","os":"ubuntu:18","ops_advertise_addr":"example.com:443"}
resize={"to":3,"flavor":"one","nodes":1,"role":"node","state_dir":"/var/lib/telekube","os":"ubuntu:18","storage_driver":"overlay2"}
shrink={"nodes":3,"flavor":"three","role":"node","os":"redhat:7"}
upgrade={"flavor":"three","nodes":3,"role":"node","os":"ubuntu:16","from":"/telekube_7.0.0.tar","service_uid":997,"service_gid":994,"storage_driver":"overlay2"}
upgrade={"flavor":"three","nodes":3,"role":"node","os":"centos:7","from":"/telekube_7.0.11.tar","service_uid":997,"service_gid":994,"storage_driver":"overlay2"}
upgrade={"flavor":"three","nodes":3,"role":"node","os":"debian:9","from":"/telekube_7.0.11.tar","service_uid":997,"service_gid":994,"storage_driver":"overlay2"}
upgrade={"flavor":"three","nodes":3,"role":"node","os":"ubuntu:18","from":"/telekube_7.0.11.tar","service_uid":997,"service_gid":994,"storage_driver":"overlay2"}
```
</details>